sonar-db/src/main/java/org/sonar/db/version/DecimalColumnDef.java

```diff
@@ -0,0 +1,117 @@
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.db.version;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+import org.sonar.db.dialect.Dialect;
+import org.sonar.db.dialect.MsSql;
+import org.sonar.db.dialect.MySql;
+import org.sonar.db.dialect.Oracle;
+import org.sonar.db.dialect.PostgreSql;
+
+import static java.util.Objects.requireNonNull;
+import static org.sonar.db.version.ColumnDefValidation.validateColumnName;
+
+public class DecimalColumnDef extends AbstractColumnDef {
+
+  private final int precision;
+  private final int scale;
+
+  private DecimalColumnDef(Builder builder) {
+    super(builder.columnName, builder.isNullable);
+    this.precision = builder.precision;
+    this.scale = builder.scale;
+  }
+
+  public static Builder newDecimalColumnDefBuilder() {
+    return new Builder();
+  }
+
+  public int getPrecision() {
+    return precision;
+  }
+
+  public int getScale() {
+    return scale;
+  }
+
+  @Override
+  public String generateSqlType(Dialect dialect) {
+    switch (dialect.getId()) {
+      case PostgreSql.ID :
+      case Oracle.ID :
+        return "NUMERIC (" + Integer.toString(precision)  + "," + Integer.toString(scale) + ")";
+      case MySql.ID :
+      case MsSql.ID :
+        return "DECIMAL (" + Integer.toString(precision)  + "," + Integer.toString(scale) + ")";
+      default :
+        return "DOUBLE";
+    }
+  }
+
+  public static class Builder {
+    @CheckForNull
+    private String columnName;
+    @CheckForNull
+    private Integer precision;
+    @CheckForNull
+    private Integer scale;
+    private boolean isNullable;
+
+
+    public Builder setColumnName(String columnName) {
+      this.columnName = validateColumnName(columnName);
+      return this;
+    }
+
+    public Builder setIsNullable(boolean isNullable) {
+      this.isNullable = isNullable;
+      return this;
+    }
+
+    public Builder setPrecision(int precision) {
+      this.precision = precision;
+      return this;
+    }
+
+    public Builder setScale(int scale) {
+      this.scale = scale;
+      return this;
+    }
+
+    public DecimalColumnDef build() {
+      validateColumnName(columnName);
```

Comment:
Just for my information, why using static methods with field as parameter ? Why not object methods ```checkPrecision()``` and ```checkScale()``` without parameters ?